datahashlasmlabscity.md

```diff
@@ -0,0 +1,101 @@
+---
+CIP: ?
+Title: New Plutus built-in `dataHash`
+Authors: Las Safin <las@mlabs.city>
+Discussions-To:
+Comments-URI:
+Status: Draft
+Type: Standards Track
+Created: 2022-02-15
+* License: Apache-2.0
+* Requires: 35
+* Replaces: 36
+---
+
+# New Plutus built-in `dataHash`
+
+## Abstract
+
+We propose adding a new built-in `dataHash` for calculating the hash of `Data` on-chain,
+in such a way that it is coherent with `txInfoData`.
+
+## Motivation
+
+This is useful for essentially "putting" `Data` into token names, the contents of which a script
+can access by having the concrete `Data` be passed in the redeemer (whether it's a minting policy
+or validator).
+
+It is also possible to verify signatures of (hashes of) `Data` with this built-in.
```

Comment:
This would require having the same implementation of `dataHash` off-chain to produce the signature or a sufficiently detailed specification on how the `Data` is serialized (and hashed) such that the system creating the signature can follow the spec.